stl/cpp_on_shore/chris/src/task_3/rb_tree.h

```diff
@@ -0,0 +1,353 @@
+#include <assert.h>
+
+#include <iostream>
+
+typedef enum { RED = 0, BLACK } Color;
+
+// 节点类型
+template <typename Type>
+struct RbTreeNode {
+  Color color;         // 颜色
+  Type key;            // 关键字
+  RbTreeNode* left;    // 左子节点
+  RbTreeNode* right;   // 右子节点
+  RbTreeNode* parent;  // 父节点
+};
+
+template <typename Type>
+class RbTree {
+ public:
```

Comment:
split node class and rbtree class may be a good design

Reply:
yes, I think I have already splited the two classes in my original design. One is the RbTreeNode class, the other is the RbTree.